# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLVertexArrayGL.cpp
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLVertexArrayGL.cpp
--- a/dom/canvas/WebGLVertexArrayGL.cpp	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGLVertexArrayGL.cpp	Wed Jan 08 22:19:16 2020 +0000
@@ -11,17 +11,14 @@
 namespace mozilla {
 
 WebGLVertexArrayGL::WebGLVertexArrayGL(WebGLContext* webgl)
-    : WebGLVertexArray(webgl, [&]() {
+    : WebGLVertexArray(webgl), mGLName([&]() {
         GLuint ret = 0;
         webgl->gl->fGenVertexArrays(1, &ret);
         return ret;
       }()) {}
 
-WebGLVertexArrayGL::~WebGLVertexArrayGL() { DeleteOnce(); }
-
-void WebGLVertexArrayGL::DeleteImpl() {
-  mElementArrayBuffer = nullptr;
-
+WebGLVertexArrayGL::~WebGLVertexArrayGL() {
+  if (!mContext) return;
   mContext->gl->fDeleteVertexArrays(1, &mGLName);
 }
 